src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1188,6 +1197,40 @@ impl<T: 'static> EventProcessor<T> {
             }
             Err(_) => (),
         }
+        match self.ime_event_receiver.try_recv() {
```

Comment:
Hi @garasubo and @murarth , I'm implementing this for Mac, should I trigger both `WindowEvent::ReceivedCharacter` and `CompositionEvent::End` at the same time, or just ignore `WindowEvent::ReceivedCharacter` during composing?

I'm not sure how XIM work, but looks like here, in `process_event`, both x event and ime event are handled?

Reply:
Thanks for get interested in this work. I don't have Mac dev environment, so that would be great.

I think we should handle both to keep compatibility, so in my implementation, both events are handled.
In XIM, IME events and receiving characters events are independent. So, I didn't touch any logic related to ReceivedCharacter event.